src/main/java/com/pp/demo/UserController.java

```diff
@@ -0,0 +1,45 @@
+package com.pp.demo;
+
+import org.springframework.stereotype.*;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletRequest;
+
+@Controller
+public class UserController {
+
+    private final UserRepository userRepository;
+
+    public UserController(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @ResponseBody
+    @GetMapping("/users")
+    public String usersList() {
+
+        return printUserList();
+    }
+
+
+    @RequestMapping("/add")
+    public String addUser(@RequestParam(name = "name", required = false) String name,
+                          @RequestParam(name = "surname", required = false) String surname,
+                          @RequestParam Integer age) {
+        User user = new User(name, surname, age);
+        if (name == null || surname == null) {
```

Comment:
Nie musisz sprawdzac nazwiska ale warto by bylo sprawdzic jeszcze czy `name.equals("")`, to tez interpretujemy jako brak imienia